pkg/tests/api_test.go

```diff
@@ -71,184 +73,274 @@ func TestGetTimestampCertChain(t *testing.T) {
+		{
+			name:         "Timestamp Query Request",
+			reqMediaType: client.TimestampQueryMediaType,
+			reqBytes:     buildTimestampQueryReq(t, []byte(testArtifact), opts),
+			nonce:        testNonce,
+			includeCerts: includeCerts,
+			hash:         testHash,
+		},
+		{
+			name:         "JSON Request",
+			reqMediaType: client.JSONMediaType,
+			reqBytes:     buildJSONReq(t, []byte(testArtifact), includeCerts, testHashStr, testNonce, ""),
+			nonce:        testNonce,
+			includeCerts: includeCerts,
+			hash:         testHash,
+		},
+	}
+
+	for _, tc := range tests {
+		url := createServer(t)
+
+		c, err := client.GetTimestampClient(url, client.WithContentType(tc.reqMediaType))
+		if err != nil {
+			t.Fatalf("unexpected error creating client: %v", err)
+		}
+
+		params := timestamp.NewGetTimestampResponseParams()
+		params.SetTimeout(10 * time.Second)
+		params.Request = io.NopCloser(bytes.NewReader(tc.reqBytes))
+
+		var respBytes bytes.Buffer
+		clientOption := func(op *runtime.ClientOperation) {
+			op.ConsumesMediaTypes = []string{tc.reqMediaType}
+		}
+		_, err = c.Timestamp.GetTimestampResponse(params, &respBytes, clientOption)
+		if err != nil {
+			t.Fatalf("unexpected error getting timestamp response: %v", err)
+		}
+
+		tsr, err := ts.ParseResponse(respBytes.Bytes())
+		if err != nil {
+			t.Fatalf("unexpected error parsing response: %v", err)
+		}
+
+		// check certificate fields
+		if len(tsr.Certificates) != 1 {
+			t.Fatalf("expected 1 certificate, got %d", len(tsr.Certificates))
+		}
+		if !tsr.AddTSACertificate {
+			t.Fatalf("expected TSA certificate")
+		}
+		// check nonce
+		if tsr.Nonce.Cmp(tc.nonce) != 0 {
+			t.Fatalf("expected nonce %d, got %d", tc.nonce, tsr.Nonce)
+		}
+		// check hash and hashed message
+		if tsr.HashAlgorithm != tc.hash {
+			t.Fatalf("unexpected hash algorithm")
+		}
+		hashedMessage := sha256.Sum256([]byte(testArtifact))
+		if !bytes.Equal(tsr.HashedMessage, hashedMessage[:]) {
+			t.Fatalf("expected hashed messages to be equal: %v %v", tsr.HashedMessage, hashedMessage)
+		}
+		// check time and accuracy
+		if tsr.Time.After(time.Now()) {
+			t.Fatalf("expected time to be set to a previous time")
+		}
+		duration, _ := time.ParseDuration("1s")
+		if tsr.Accuracy != duration {
+			t.Fatalf("expected 1s accuracy, got %v", tsr.Accuracy)
+		}
+		// check serial number
+		if tsr.SerialNumber.Cmp(big.NewInt(0)) == 0 {
+			t.Fatalf("expected serial number, got 0")
+		}
+		// check ordering and qualified defaults
+		if tsr.Qualified {
+			t.Fatalf("tsr should not be qualified")
+		}
+		if tsr.Ordering {
+			t.Fatalf("tsr should not be ordered")
+		}
+		// check policy OID default
+		if !tsr.Policy.Equal(asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 2}) {
+			t.Fatalf("unexpected policy ID")
+		}
+		// check for no extensions
+		if len(tsr.Extensions) != 0 {
+			t.Fatalf("expected 0 extensions, got %d", len(tsr.Extensions))
+		}
 	}
 }
 
 func TestGetTimestampResponseWithExtsAndOID(t *testing.T) {
```

Comment:
Looking back at the timestamp-cli timestamp command, it doesn't look like it supports providing extra extensions. Since this test includes setting extra extensions, are we interested in adding a flag for extra extensions as well?

Reply:
Sure, do you want to file an issue for this? I think it'd also be fine to say that the average CLI user won't care about extensions and we'd also support extensions via the API.

Reply:
Supporting extensions just through the API sounds fine to me.